### PR TITLE
Fix XSD validation error for generated polygons by adding a unique id…

### PR DESCRIPTION
**Pull request by jonescc**
… to each polygon

### CODE REVIEW

**Comment by danfruehauf** [comment]
Please explain why this is required.

**Comment by jonescc** [comment]
When validating MCP 2.0 metadata against the MCP 2.0 xsd schema we get errors because polygons created by the iUpdateSpatialExtent component do not include gml:id.

File: extension/imos--1.0.sql [hunk 1]
```diff
@@ -271,6 +271,28 @@ END;
 $BODY$
 LANGUAGE plpgsql; 
 
+-- Function to add ids to polygons in provided gml
+
+CREATE FUNCTION add_id_to_polygons( p_gml text )
```
Comment (danfruehauf) [inline extension/imos--1.0.sql]: What's the naming convention here. Is it supposed to be `add_id_to_polygons` or `addIdToPolygons`?

**Comment by danfruehauf** [comment]
> When validating MCP 2.0 metadata against the MCP 2.0 xsd schema we get errors because polygons created by the iUpdateSpatialExtent component do not include gml:id.

That means the talend component needs changing? Or is it using this library?

If so, then it means we need to trigger all harvesters to regenerate the bounding polygons they post to geonetwork?

File: extension/imos--1.0.sql (hunk 1 above)
Comment (jonescc) [inline extension/imos--1.0.sql]: Well its a bit of a mixed bag in postgis - but I think underscores are winning.  I've switched to that as postgres is case insensitivce when it comes to function naming - so its more readable.

**Comment by jonescc** [comment]
I changed it in postgres so I wouldn't need to change the talend component and recompile everything.  we can also just do a one-off sql update rather than running all the harvesters

**Comment by jonescc** [comment]
Or just leave it to fix itself over time

File: extension/imos--1.0.sql (hunk 1 above)
Comment (danfruehauf) [inline extension/imos--1.0.sql]: Fine with me.

**Comment by danfruehauf** [comment]
> Or just leave it to fix itself over time

Not all harvesters are triggered automatically - FYI. I prefer to bulk SQL update it. Perhaps even paste the required SQL here.

And another question, will the new changes actually be deployed with a new git revision? @julian1?

**Comment by jonescc** [comment]
Yes they are - go on try it in vagrant.  Good work there @julian1

**Comment by danfruehauf** [comment]
@jonescc Don't forget you need to run the bulk update when appropriate.

**Comment by jonescc** [comment]
Yeah looking at it now.  Thanks this PR did not have to wait for that.

**Comment by jonescc** [comment]
OK, so it looks like chef isn't redeploying the extension when it's changed like it used to.   I'll raise a chef issue.

**Pull request merged**